spec/javascripts/components/table-filter-spec.js

```diff
@@ -0,0 +1,121 @@
+/* eslint-env jasmine */
+/* global GOVUK */
+
+describe('Table Filter component', function () {
+  'use strict'
+
+  var element
+  var filter
+  var filterCount
+  var searchInput
+  var rows
+  var message
+  var tableFilter
+  var hiddenClass
+  var event
+  var FIXTURE
+
+  beforeEach(function () {
+    FIXTURE =
+      '<div data-module="table-filter">' +
+        '<div class="js-gem-c-table__filter govuk-!-display-none">' +
+          '<input name="filter">' +
+          '<p id="filterCount" class="js-filter-count" data-count-text="Number of filtered rows"></p>' +
+        '</div>' +
+        '<table>' +
+          '<thead></thead>' +
+          '<tbody class="govuk-table__body">' +
+            '<tr class="js-govuk-table__row">' +
+              '<td>bcd</td>' +
+              '<td>efg</td>' +
+              '<td>hio</td>' +
+            '</tr>' +
+            '<tr class="js-govuk-table__row">' +
+              '<td>klm</td>' +
+              '<td>nop</td>' +
+              '<td>£35</td>' +
+            '</tr>' +
+          '</tbody>' +
+        '</table>' +
+        '<p class="js-gem-c-table__message govuk-!-display-none">That search returns no results.</p>' +
+      '</div>'
+
+    window.setFixtures(FIXTURE)
+  })
+
+  describe('When the component is intialised', function () {
+    beforeEach(function () {
+      element = document.querySelector('[data-module="table-filter"]')
+      filter = document.querySelector('.js-gem-c-table__filter')
+      searchInput = document.querySelector('input')
+      tableFilter = new GOVUK.Modules.TableFilter(element)
+      event = new window.Event('input')
+      hiddenClass = tableFilter.hiddenClass
+
+      spyOn(tableFilter, 'updateRows')
+      tableFilter.init()
+    })
+
+    it('Displays the filter section', function () {
+      expect(filter.classList.contains(hiddenClass)).toBe(false)
+    })
+
+    it('Sets up the event listener', function () {
+      searchInput.dispatchEvent(event)
+      expect(tableFilter.updateRows).toHaveBeenCalledTimes(1)
+    })
+  })
+
+  describe('When the user inputs a value', function () {
+    beforeEach(function () {
```

Comment:
Is there any significant difference in the two `beforeEach` functions you've defined here? They look broadly similar although the second one has some extra bits (but nothing that contradicts the other?). If so, you could write one and put it at the top of the file for all tests?